lib/engine/step/g_18_mex/buy_sell_par_shares.rb

```diff
@@ -15,16 +15,40 @@ def can_sell?(entity, bundle)
         end
 
         def can_gain?(entity, bundle)
-          return super if bundle.corporation != @game.ndm || bundle&.percent != 5
+          super && !attempt_ndm_action_on_unavailable?(bundle)
+        end
+
+        # Check if it is possible to buy an NdM IPO or Pool share of 10%
+        # when player swaps in a 5% share.
+        def swap_buy(player, corporation, ipo_or_pool_share)
+          return if @game.ndm != corporation || ipo_or_pool_share.percent != 10
+
+          swap_share = player.shares_of(corporation).find { |s| s.percent == 5 }
+          return unless swap_share
+
+          # If we were allowed to buy another 5% then swap is OK.
+          # We test that a reduced buy of 5% would be allowed.
+          updated_share = ipo_or_pool_share.dup
+          updated_share.percent -= 5
+          can_buy?(player, updated_share) ? swap_share : nil
+        end
+
+        # Check if it is possible to sell an NdM bundle if player swap 5% share from pool
+        def swap_sell(player, corporation, bundle, pool_share)
+          return if @game.ndm != corporation || pool_share.percent != 5 || bundle.percent == 5
 
-          # NdM 5% shares does not affect cert limit
-          bundle.corporation.holding_ok?(entity, bundle.percent)
+          # If we were allowed to buy another 5% then swap is OK. Test this by
+          # creating a cloned bundle where one of the shares has its percentage reduced
+          # by 5. This way we can test if the swap will not exceed market limit of 50%.
+          updated_shares = bundle.shares.map(&:dup)
```

Comment:
why do you need to dup these shares?

Reply:
The bundle sent in is the one that are to be sold, so I do not want to affect it, but create a clone. And for that clone I reduce the percentage with 5 (which is the size of NdM 5% shares to be swapped with).ö
Anyway, I refactored this part of the code to use a helper method. Used in both the can_buy? and can_sell? tests.